catalog/app/containers/Bucket/CopyButton.js

```diff
@@ -0,0 +1,142 @@
+import * as R from 'ramda'
+import * as React from 'react'
+import * as M from '@material-ui/core'
+import * as Lab from '@material-ui/lab'
+
+import * as AWS from 'utils/AWS'
+import AsyncResult from 'utils/AsyncResult'
+import { useData } from 'utils/Data'
+
+import * as requests from './requests'
+
+function MenuPlaceholder() {
+  const t = M.useTheme()
+
+  return (
+    <M.Box minWidth={t.spacing(22)}>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(4)} width="100%" />
+      </M.MenuItem>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(4)} width="100%" />
+      </M.MenuItem>
+      <M.MenuItem disabled>
+        <Lab.Skeleton height={t.spacing(4)} width="100%" />
+      </M.MenuItem>
+    </M.Box>
+  )
+}
+
+function MenuItem({ item, onClick }) {
+  return (
+    <M.MenuItem onClick={React.useCallback(() => onClick(item), [item, onClick])}>
+      <M.ListItemText primary={item.name} secondary={item.url} />
+    </M.MenuItem>
+  )
+}
+
+const BucketsListFetcher = React.forwardRef(({ bucket, children }, ref) => {
+  const s3 = AWS.S3.use()
+  const data = useData(requests.workflowsList, { s3, bucket })
+  const res = data.case({
```

Comment:
I'm not familiar with how to use this API properly yet. Do you mean it can be safe to do:

```
const data = useData(…)
return children({
  ref,
  ...data,
})
```

Reply:
Code from my comment doesn't work with exception `Invariant Violation: app/utils/tagged/case: must be called with an instance of type, otherwise cases must include the __ placeholder`.
So please elaborate on how can I change it

Reply:
useData returns an object like `{ result, fetch, case }` ([the code](https://github.com/quiltdata/quilt/blob/master/catalog/app/utils/Data.js#L56)), you need `.result` (it's an instance of AsyncResult), e.g. `return children({ ref, ...data.result })`